### PR TITLE
Added MGXS Reaction Rate Tally Attribute

### PR DESCRIPTION
**Pull request by wbinventor**
This PR adds a new `_rxn_rate_tally` attribute and `rxn_rate_tally` property to the `openmc.mgxs.MGXS` class. This can be useful for some downstream data processing which need access to the derived tally used to represent the multi-group reaction rate (*i.e.*, the "transport" rate for a transport cross section).

### CODE REVIEW

**Comment by paulromano** [comment]
Would it make sense override the `rxn_rate_tally` property for each subclass of MGXS to generate `_rxn_rate_tally` and then have the `MGXS.xs_tally` property do reaction rate / flux and call compute_xs? Seems then you could avoid having to call `compute_xs()` for the subclasses.

**Comment by wbinventor** [comment]
I think that would indeed be possible and would simplify the data flow. The only complication would be `Chi` which does not divide by the flux, but it could simply over-ride the parent classes `xs_tally` property.

**Comment by wbinventor** [comment]
I refactored the `MGXS` class to calculate cross sections on the fly when the `xs_tally` instance attribute is accessed. The diff for this PR is larger now, but I really like these changes as it makes the cross section calculation happen on-the-fly in a way that mirrors OTF tally data loading into the `StatePoint`.

File: openmc/mgxs/mgxs.py [hunk 1]
```diff
@@ -1748,23 +1747,22 @@ def correction(self, correction):
...
-            scatter_p1 = self.tallies['scatter-P1']
-            scatter_p1 = scatter_p1.get_slice(scores=['scatter-P1'])
-            energy_filter = self.tallies['scatter'].find_filter('energy')
-            energy_filter = copy.deepcopy(energy_filter)
-            scatter_p1 = scatter_p1.diagonalize_filter(energy_filter)
-            rxn_tally = self.tallies['scatter'] - scatter_p1
-        else:
-            rxn_tally = self.tallies['scatter']
+    @property
+    def rxn_rate_tally(self):
```
Comment (paulromano) [inline openmc/mgxs/mgxs.py]: Move this above the setters?

**Comment by paulromano** [comment]
I really like it too; nice!

File: openmc/mgxs/mgxs.py (hunk 1 above)
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: Agreed and done

**Pull request merged**